src/compiler/typescript_to_rust/azle_generate_rearchitecture/src/main.rs

```diff
@@ -104,11 +105,20 @@ fn main() -> Result<(), String> {
             let js_function_name = &canister_method.name;
             let is_composite = canister_method.composite.unwrap_or(false);
 
+            let (guard_attribute, guard_function) = if let Some(guard_name) = &canister_method.guard_name
+            {
+                get_guard_token_stream(guard_name)
+            } else {
+                (quote!(), quote!())
+            };
+
             quote! {
-                #[ic_cdk_macros::query(manual_reply = true, composite = #is_composite)]
+                #[ic_cdk_macros::query(manual_reply = true, composite = #is_composite #guard_attribute)]
```

Comment:
Is there supposed to be comma between composite and #guard_attribute? I'm sure you did it right but it caught my eye so just double check.

Reply:
It's correct, the comma is in the guard_attribute